enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,293 @@
+- the required capabilities is a configuration option of the cluster version operator (e.g. `--required-capabilities` flag).
+- the cluster version operator should [implicitly enable](https://github.com/openshift/cluster-version-operator/blob/master/pkg/cvo/status.go#L161-L165) all the provided required capabilities so that they appear in the status of `ClusterVersion` object.
+- the required capabilities should be added to [the cluster version operator manifests](https://github.com/openshift/cluster-version-operator/blob/master/install/0000_00_cluster-version-operator_03_deployment.yaml) only.
+- the ingress capability should be required.
+
+This approach should make any required capability read only on the standalone OpenShift. Also, it doesn't let the required capabilities to be set on HyperShift because HyperShift doesn't use the cluster version operator's deployment manifest but [manages the operator's deployment directly](https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L318).
+
+#### Ingress operator APIs
+
+The following APIs are not available if the ingress capability is disabled:
+- CRDs which are part of the cluster operator's manifest payload:
+  - [IngressController](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-custom-resource-definition.yaml)
+  - [DNSRecord](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-custom-resource-definition-internal.yaml)
+- CRDs which are created at runtime if `gateway-api` feature gate is set:
+  - [GatewayClass](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gatewayclasses.yaml)
+  - [Gateway](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gateways.yaml)
+  - [HTTPRoute](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_httproutes.yaml)
+
+#### Related APIs
+
+The following APIs are left intact (see [Add Route API to the ingress capability](#add-route-api-to-the-ingress-capability) to know why) if the ingress capability is disabled:
+- [Route API](https://docs.openshift.com/container-platform/4.13/rest_api/network_apis/route-route-openshift-io-v1.html). Implemented in two places: OpenShift API as [CRD](https://github.com/openshift/api/blob/master/route/v1/route.crd.yaml) and in [openshift-api-server](https://github.com/openshift/openshift-apiserver/tree/master/pkg/route).
+- [Ingress Config API](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html). Implemented in OpenShift API as [CRD](https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_ingress.crd.yaml).
+
+However the implementation of these APIs (or parts of them) depend on the ingress operator:
```

Comment:
might be worth stating the implication explicitly:

if someone creates a Route resource, nothing will happen.

Also what will happen if someone creates an Ingress resource that normally would have been reconciled into a Route?  I assume that functionality is also missing when the Ingress capability is disabled?  (Even if it did create the Route, the Route would do nothing, but it would be good to understand if the Route object will be created to back the Ingress object or not)

Reply:
Done. The links to the dedicated risks and drawbacks sections are added. I had to add the "ingress to route controller" risk section and slightly update the drawbacks for this.

Reply:
thanks, resolving this one.